homeassistant/components/homekit/type_lights.py

```diff
@@ -282,7 +282,11 @@ def async_update_state(self, new_state: State) -> None:
                 hue, saturation = color_temperature_to_hs(color_temp)
             elif color_mode == ColorMode.WHITE:
                 hue, saturation = 0, 0
-            elif hue_sat := attributes.get(ATTR_HS_COLOR):
+            elif (
+                (hue_sat := attributes.get(ATTR_HS_COLOR))
+                and isinstance(hue_sat, (list, tuple))
```

Comment:
What about using `Sequence` in the instance check?

Reply:
It should only ever be a `tuple` but I've seen custom components use `list`. I was trying to be a bit permissive but not allow too much garbage.